lasagne/layers/prelu.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import theano.tensor as T
+
+from .. import init
+from .. import nonlinearities
+
+from .base import Layer
+
+__all__ = [
+    "PReLULayer"
+]
+
+class PReLULayer(Layer):
+    """
+    A layer used to implement Parametric Rectified Linear Units (PReLU).
+    This layer simply applies the parametric nonlinearity to the previous
+    layer's output. 
+
+    Parameters
+    -----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+
+    alpha : double or None
+        The initial "leakiness" of the rectifier which is then learned.
+        If argument not provided, initial leakiness will be set to 0.25.
+
+    unique_chan_param : boolean or None
```

Comment:
For the tuple case are we thinking of (False, True, True, False) if we want
to untie across dim 1 and 2?



Greg

On Fri, May 15, 2015 at 8:28 AM, Jan Schlüter <notifications@github.com>
wrote:

> In lasagne/layers/prelu.py
> <https://github.com/Lasagne/Lasagne/pull/247#discussion_r30404210>:
>
> > +class PReLULayer(Layer):
> > +    """
> > +    A layer used to implement Parametric Rectified Linear Units (PReLU).
> > +    This layer simply applies the parametric nonlinearity to the previous
> > +    layer's output.
> > +
> > +    Parameters
> > +    -----------
> > +    incoming : a :class:`Layer` instance or a tuple
> > +        The layer feeding into this layer, or the expected input shape.
> > +
> > +    alpha : double or None
> > +        The initial "leakiness" of the rectifier which is then learned.
> > +        If argument not provided, initial leakiness will be set to 0.25.
> > +
> > +    unique_chan_param : boolean or None
>
>  In my own throwaway implementation I used "untie_channels", a bit like
> the "untie_biases" argument for the conv layers.
>
> Following that logic, it should be called untie_alphas, not untie_channels.
> Also see #249 <https://github.com/Lasagne/Lasagne/issues/249>, maybe we
> should allow specifying exactly which axes to share alphas/biases over.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Lasagne/Lasagne/pull/247/files#r30404210>.
>